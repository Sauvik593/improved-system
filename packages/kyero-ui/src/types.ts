import type { LinkHTMLAttributes } from 'react';

export type BaseLinkProps =
  | {
      to: string;
    }
  | {
      to: {
        search?: string;
        pathname?: string;
      };
    };

export type LinkProps = BaseLinkProps & LinkHTMLAttributes<HTMLAnchorElement>;

export type LinkComponentProps = {
  className: string;
  children: React.ReactNode;
} & LinkProps;
