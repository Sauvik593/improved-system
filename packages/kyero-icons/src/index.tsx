export * from './AnimatedSpinner';
export * from './ArrowUp';
export * from './Account';
export * from './ArrowDown';
export * from './Advertise';
export * from './ArrowBackward';
export * from './ArrowForward';
export * from './AlertInfo';
export * from './Avatar';
export * from './BoostRocket';
export * from './Camera';
export * from './Cart';
export * from './Close';
export * from './Chat';
export * from './Checked';
export * from './ChevronDown';
export * from './ChevronLeft';
export * from './Dashboard';
export * from './DeleteIcon';
export * from './Edit';
export * from './Email';
export * from './Eye';
export * from './Floorplans';
export * from './Forbidden';
export * from './Furniture';
export * from './Gate';
export * from './Globe';
export * from './Golf';
export * from './Hamburger';
export * from './HouseIcon';
export * from './JoinLogo';
export * from './List';
export * from './Logout';
export * from './LogoWide';
export * from './Imports';
export * from './InfoIcon';
export * from './Kyero';
export * from './NextIcon';
export * from './Outside';
export * from './Parking';
export * from './Photo';
export * from './Plus';
export * from './Pool';
export * from './PrevIcon';
export * from './Processing';
export * from './Properties';
export * from './Quote';
export * from './RemoteViewing';
export * from './Reply';
export * from './School';
export * from './Search';
export * from './ShareIcon';
export * from './Star';
export * from './SunMark';
export * from './Temperature';
export * from './Tick';
export * from './Town';
export * from './QuestionMark';
export * from './WavyArrow';
export * from './Views';
export * from './VirtualTour';
export * from './XIcon';
