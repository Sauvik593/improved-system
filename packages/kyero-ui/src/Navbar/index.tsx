export * from './Navbar';
export * from './NavElement';
