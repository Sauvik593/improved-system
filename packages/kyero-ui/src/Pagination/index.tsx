import cn from 'classnames';
import { Page } from './Page';
import { Button } from './Button';
import { usePagination } from './usePagination';

import type { LinkComponentProps } from '../types';

export interface PaginationProps {
  totalPages: number;
  currentPage: number;
  pathname: string;
  nextPage: number | null;
  prevPage: number | null;
  LinkComponent?: (props: LinkComponentProps) => React.ReactElement;
  search: string;
}

const DefaultComponent = ({ to, children, className }: LinkComponentProps) => {
  const path = typeof to === 'string' ? to : `${to?.pathname}?${to?.search}`;
  return (
    <a href={path} className={cn(className)}>
      {children}
    </a>
  );
};

export const Pagination = ({
  currentPage,
  totalPages,
  pathname,
  prevPage,
  nextPage,
  LinkComponent = DefaultComponent,
  search,
}: PaginationProps) => {
  const { showPrevDots, showNextDots, pagesArray } = usePagination(search, currentPage, totalPages);
  return (
    <nav className="mt-4">
      <ul className="flex items-center gap-2">
        <Button
          page={prevPage}
          pathname={pathname}
          search={search}
          type="prev"
          LinkComponent={LinkComponent}
          testId="pagination-prev"
        />
        {totalPages >= 2 && (
          <Page
            page={1}
            currentPage={currentPage}
            pathname={pathname}
            search={search}
            LinkComponent={LinkComponent}
            testId="pagination-firstpage"
          />
        )}
        {showPrevDots && (
          <li className="font-bold" data-testid="pagination-more-prev-dots">
            ...
          </li>
        )}
        {pagesArray.map((page) => {
          return (
            <Page
              key={page}
              page={page}
              currentPage={currentPage}
              pathname={pathname}
              search={search}
              LinkComponent={LinkComponent}
              testId={`pagination-page-${page}`}
            />
          );
        })}
        {showNextDots && (
          <li className="font-bold" data-testid="pagination-more-next-dots">
            ...
          </li>
        )}
        <Page
          page={totalPages}
          currentPage={currentPage}
          pathname={pathname}
          search={search}
          LinkComponent={LinkComponent}
          testId="pagination-lastpage"
        />
        <Button
          page={nextPage}
          pathname={pathname}
          search={search}
          type="next"
          testId="pagination-next"
          LinkComponent={LinkComponent}
        />
      </ul>
    </nav>
  );
};
