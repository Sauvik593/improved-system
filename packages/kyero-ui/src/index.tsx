export * from './Alert';
export * from './ArrowLink';
export * from './Badge';
export * from './Button';
export * from './Checkbox';
export * from './CheckboxList';
export * from './FiltersField';
export * from './Footer';
export * from './LanguageSwitcher';
export * from './Navbar';
export * from './Pagination';
export * from './Radio';
export * from './RadioList';
export * from './SearchInput';
export * from './Select';
export * from './Switch';
export * from './Tooltip';
