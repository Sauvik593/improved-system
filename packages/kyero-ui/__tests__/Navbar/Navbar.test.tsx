import renderer from 'react-test-renderer';
import { render, queryByTestId } from '@testing-library/react';
import { Avatar, Dashboard } from '@kyero/icons';

import type { NavElementProps, NavbarProps } from '../../src';
import { Navbar, NavElement } from '../../src';

const DummyLink = (props: NavElementProps['linkProps'] & { children: React.ReactNode }) => {
  return (
    <div data-testid="dummy-link" className={props.className || ''}>
      {props.children}
    </div>
  );
};

const LINKS = [
  {
    title: 'Test1',
    icon: <Dashboard />,
    active: true,
    linkProps: { to: '/' },
    component: DummyLink,
  },
  { title: 'Test2', icon: <Avatar />, linkProps: { to: '/' } },
];

const BOTTOM_PROPS = {
  bottomProps: {
    title: 'Testing title',
    subtitle: 'Testing subtitle',
    imgSrc: 'test.jpg',
  },
};

const renderNav = (props: Partial<NavbarProps> = {}) => (
  <Navbar {...props}>
    {LINKS.map((link) => (
      <NavElement {...link} key={link.title} />
    ))}
  </Navbar>
);

describe('<Navbar />', () => {
  it('should render correctly', () => {
    const navbar = renderer.create(renderNav()).toJSON();
    expect(navbar).toMatchSnapshot();
  });

  describe('rendering list', () => {
    it('should render custom component in a correct place', () => {
      const { getAllByTestId } = render(renderNav());

      const navElements = getAllByTestId('nav-element');
      const [firstElement, secondElement] = navElements;

      expect(navElements).toHaveLength(2);
      expect(queryByTestId(firstElement, 'dummy-link')).toBeTruthy();
      expect(queryByTestId(secondElement, 'dummy-link')).toBeFalsy();
    });
  });

  describe('rendering footer', () => {
    it('should render correctly', () => {
      const navbar = renderer.create(renderNav(BOTTOM_PROPS)).toJSON();
      expect(navbar).toMatchSnapshot();
    });

    it('should render footer correctly', () => {
      const { getByTestId } = render(renderNav(BOTTOM_PROPS));

      expect(getByTestId('nav-bottom-user')).toBeTruthy();
    });
  });
});
