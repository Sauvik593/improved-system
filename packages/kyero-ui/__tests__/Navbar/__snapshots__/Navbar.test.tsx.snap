// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> rendering footer should render correctly 1`] = `
<section
  className="bg-sierra-night-100 flex h-full min-h-screen flex-col "
>
  <div
    className="flex p-5"
  >
    <picture
      className=""
    >
      <svg
        className="text-white"
        fill="none"
        viewBox="0 0 152 40"
        width={106}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M115.218 33.268c-1.532 0-2.482-1.063-2.482-2.46V19.87c0-3.737.827-6.016 2.973-8.082 1.808-1.762 4.904-2.795 7.172-2.643 1.38.091 2.391.973 2.33 2.43-.031 1.429-1.104 2.34-2.483 2.28-1.226-.03-2.636.486-3.464 1.306-1.042 1.002-1.532 2.34-1.532 4.557v11.059c-.031 1.428-.981 2.491-2.514 2.491z"
          fill="currentColor"
        />
        <path
          d="M62.16 29.896c-1.378-4.497-4.014-7.2-6.007-8.78-.95-.76-2.237-1.61-3.586-2.218 1.196-1.064 5.303-4.74 6.008-5.469.98-1.033 1.072-2.491.122-3.494-1.042-1.154-2.482-1.033-3.617-.121-.766.638-5.517 5.225-7.846 7.474V2.522c0-1.428-.981-2.521-2.514-2.521-1.563-.03-2.544 1.063-2.544 2.521v28.194c0 1.428.981 2.522 2.544 2.522 1.563 0 2.514-1.094 2.514-2.522v-8.385c.214 0 .368.03.613.03 2.82.304 7.54 3.22 9.41 9.054.52 1.58 1.992 2.066 3.218 1.701 1.38-.395 2.176-1.701 1.686-3.22z"
          fill="currentColor"
        />
        <path
          d="M83.923 9.328c-1.379-.578-2.666-.03-3.218 1.306-.061.122-4.169 9.844-6.682 15.494l-7.05-15.555c-.552-1.215-1.93-1.883-3.34-1.245-1.38.638-1.779 2.005-1.258 3.311l8.92 19.292c-.766 1.367-1.257 1.914-1.931 2.34-.49.303-1.073.516-2.697.516-1.318-.03-2.269.85-2.269 2.37 0 1.489 1.135 2.34 2.453 2.4 2.084.122 3.74-.243 5.21-1.185 2.177-1.367 3.127-3.342 4.66-6.532 1.44-3.008 8.153-18.563 8.46-19.292.612-1.306.122-2.643-1.258-3.22z"
          fill="currentColor"
        />
        <path
          d="M109.027 17.379c-.981-4.74-5.517-8.416-10.881-8.416-6.1 0-11.556 4.74-11.556 12.244 0 3.402 1.195 6.592 3.341 8.84 2.115 2.219 5.303 3.373 8.521 3.373 2.759 0 5.425-.668 7.295-1.914 1.104-.73 1.533-1.975.92-3.16-.705-1.367-2.085-1.64-3.249-.972-1.533.82-2.912 1.489-5.027 1.489-2.973 0-5.487-1.762-6.468-4.83 3.587-.7 12.384-2.431 14.315-2.826 2.145-.486 3.218-1.854 2.789-3.828zm-17.38 2.126c.552-3.463 2.82-5.985 6.53-5.985 2.543 0 4.72 1.459 5.67 3.585l-12.2 2.4z"
          fill="currentColor"
        />
        <path
          d="M139.433 8.75c-7.08 0-12.567 5.439-12.567 12.457 0 7.018 5.487 12.456 12.567 12.456S152 28.225 152 21.207 146.513 8.75 139.433 8.75zm0 20.295c-4.261 0-7.571-3.403-7.571-7.838 0-4.436 3.28-7.839 7.571-7.839 4.261 0 7.571 3.403 7.571 7.839 0 4.405-3.31 7.838-7.571 7.838z"
          fill="currentColor"
        />
        <path
          d="M17.594 9.176a8.992 8.992 0 00-4.506 1.215c.399 1.246.613 2.613.613 3.98 0 .73-.061 1.458-.184 2.157a15.454 15.454 0 015.67 9.995 8.682 8.682 0 007.235-8.567c0-4.861-3.955-8.78-8.828-8.78z"
          fill="#F5B333"
        />
        <path
          d="M11.127 2.036c.183-.334.46-.486 1.042-.699.582-.212.888-.243 1.226-.06.337.152.49.364.674.88.123.396.95 3.403 1.042 3.98.184.79-.214 1.277-.889 1.49-.674.242-1.256.06-1.594-.7-.245-.546-1.379-3.463-1.532-3.858-.123-.456-.123-.729.03-1.033z"
          fill="#F5B333"
        />
        <path
          d="M21.793 1.307c.337-.152.675-.121 1.257.061.582.182.858.334 1.042.668.184.335.184.578 0 1.094-.153.395-1.257 3.312-1.532 3.828-.338.73-.92.912-1.594.699-.675-.213-1.073-.668-.859-1.489.123-.577.92-3.615 1.043-4.01.183-.486.337-.699.643-.85z"
          fill="#F5B333"
        />
        <path
          d="M30.866 6.928c.368.06.613.303.98.79.369.486.491.76.46 1.124-.06.364-.214.577-.643.911-.337.243-2.973 1.944-3.494 2.218-.705.395-1.288.213-1.686-.365-.43-.577-.46-1.154.184-1.701.46-.395 2.88-2.37 3.218-2.643.398-.304.644-.395.98-.334z"
          fill="#F5B333"
        />
        <path
          d="M34.881 16.771c.276.274.307.577.337 1.215.031.608-.06.912-.306 1.185-.245.274-.49.334-1.073.365-.429 0-3.555-.183-4.168-.243-.828-.091-1.165-.578-1.134-1.276 0-.699.306-1.215 1.164-1.276.613-.061 3.74-.243 4.169-.274.49-.03.766.061 1.011.304z"
          fill="#F5B333"
        />
        <path
          d="M32.276 27.04c.03.365-.092.669-.46 1.155-.337.516-.582.698-.95.79-.368.06-.613-.03-1.073-.335-.337-.243-2.79-2.217-3.218-2.612-.613-.547-.583-1.155-.153-1.702.398-.577.98-.79 1.685-.364.521.303 3.188 1.974 3.525 2.217.43.304.583.517.644.851z"
          fill="#F5B333"
        />
        <path
          d="M24.061 33.846c-.184.334-.46.486-1.042.698-.582.213-.889.243-1.226.061-.337-.152-.49-.364-.674-.881-.123-.395-.95-3.403-1.042-3.98-.184-.79.214-1.276.888-1.489.675-.243 1.257-.06 1.594.7.246.546 1.41 3.463 1.533 3.858.153.455.123.729-.03 1.033z"
          fill="#F5B333"
        />
        <path
          d="M11.617 17.288c.245-.942.368-1.914.368-2.887 0-4.192-2.269-7.838-5.67-9.843a17.302 17.302 0 00-1.87 1.792c3.003 1.489 5.057 4.527 5.057 8.051 0 .547-.061 1.094-.153 1.641-.521-.213-1.042-.425-1.594-.577.061-.335.092-.699.092-1.033 0-3.008-1.84-5.59-4.475-6.745a18.57 18.57 0 00-1.318 2.096c1.93.669 3.34 2.492 3.34 4.649 0 .182 0 .364-.03.547a13.635 13.635 0 00-1.594-.091h-.092c.03-.152.03-.304.03-.456a3.204 3.204 0 00-2.39-3.1c-.307.73-.582 1.49-.766 2.28.398.06.735.394.735.82 0 .455-.367.85-.858.85-.092 0-.153 0-.214-.03 0 .03 0 .091-.031.122-.03.121-.03.212-.03.334 0 .06-.031.121-.031.182A14.694 14.694 0 000 17.956v.03a11.047 11.047 0 013.77-.668c6.253 0 11.341 5.043 11.341 11.241 0 2.309-.705 4.466-1.93 6.259.857.212 1.716.364 2.635.455a13.51 13.51 0 001.778-6.744c0-4.649-2.39-8.78-5.977-11.241z"
          fill="#9FACD9"
        />
        <path
          d="M3.77 19.445a9.481 9.481 0 00-3.617.729c.123.85.276 1.67.49 2.46a6.692 6.692 0 013.127-.759c3.74 0 6.774 3.008 6.774 6.715a6.477 6.477 0 01-1.747 4.466c.736.425 1.471.79 2.268 1.093 1.196-1.55 1.931-3.463 1.931-5.56-.03-5.043-4.138-9.144-9.226-9.144z"
          fill="#9FACD9"
        />
        <path
          d="M3.77 24.002a4.48 4.48 0 00-2.39.668c.306.76.674 1.489 1.103 2.157a2.178 2.178 0 011.287-.425c1.196 0 2.176.972 2.176 2.157a2.08 2.08 0 01-.827 1.671 23.14 23.14 0 001.839 1.61 4.57 4.57 0 001.41-3.28c0-2.522-2.054-4.558-4.598-4.558z"
          fill="#9FACD9"
        />
      </svg>
    </picture>
  </div>
  <nav>
    <ul
      className="text-white"
    >
      <li
        className="flex items-center"
        data-testid="nav-element"
      >
        <div
          className="w-full px-4 py-3 flex hover:bg-sierra-night-80 focus:bg-sierra-night-80 bg-sierra-night-80"
          data-testid="dummy-link"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.5714 11.2857H14.8571C14.4626 11.2857 14.1428 11.6055 14.1428 12V20.5715C14.1428 20.9659 14.4626 21.2857 14.8571 21.2857H20.5714C20.9659 21.2857 21.2857 20.9659 21.2857 20.5715V12C21.2857 11.6055 20.9659 11.2857 20.5714 11.2857Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
            <path
              d="M20.5714 2.71429H14.8571C14.4626 2.71429 14.1428 3.03409 14.1428 3.42858V6.30001C14.1428 6.6945 14.4626 7.01429 14.8571 7.01429H20.5714C20.9659 7.01429 21.2857 6.6945 21.2857 6.30001V3.42858C21.2857 3.03409 20.9659 2.71429 20.5714 2.71429Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
            <path
              d="M9.14293 2.71429H3.42864C3.03415 2.71429 2.71436 3.03409 2.71436 3.42858V12C2.71436 12.3945 3.03415 12.7143 3.42864 12.7143H9.14293C9.53742 12.7143 9.85721 12.3945 9.85721 12V3.42858C9.85721 3.03409 9.53742 2.71429 9.14293 2.71429Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
            <path
              d="M9.14293 16.9857H3.42864C3.03415 16.9857 2.71436 17.3055 2.71436 17.7V20.5714C2.71436 20.9659 3.03415 21.2857 3.42864 21.2857H9.14293C9.53742 21.2857 9.85721 20.9659 9.85721 20.5714V17.7C9.85721 17.3055 9.53742 16.9857 9.14293 16.9857Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
          </svg>
          <span
            className="ml-2"
          >
            Test1
          </span>
        </div>
      </li>
      <li
        className="flex items-center"
        data-testid="nav-element"
      >
        <a
          className="w-full px-4 py-3 flex hover:bg-sierra-night-80 focus:bg-sierra-night-80"
          href="/"
        >
          <svg
            fill="none"
            height={19}
            viewBox="0 0 19 20"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9.24875"
              cy="9.24875"
              fill="#232559"
              r="9.24875"
            />
            <mask
              height="17"
              id="mask0_22_2320"
              maskUnits="userSpaceOnUse"
              style={
                {
                  "maskType": "alpha",
                }
              }
              width="14"
              x="2"
              y="3"
            >
              <path
                d="M12.4645 6.90497C12.4645 8.73556 11.0101 10.9933 9.13163 10.9933C7.25311 10.9933 5.85938 8.73556 5.85938 6.90497C5.85938 5.07438 7.31371 3.54889 9.19222 3.54889C11.0707 3.54889 12.4645 5.01336 12.4645 6.90497Z"
                fill="#B3B3B3"
              />
              <path
                d="M15.4338 16.546C15.4338 16.7901 12.1615 19.0479 9.13165 19.0479C6.22298 18.9868 2.89014 16.9122 2.89014 16.546C2.89014 13.8002 6.46537 11.6035 9.19225 11.6035C11.9191 11.6035 15.4338 13.8002 15.4338 16.546Z"
                fill="#B3B3B3"
              />
            </mask>
            <g
              mask="url(#mask0_22_2320)"
            >
              <circle
                cx="9.24863"
                cy="9.24881"
                fill="#EDEDF0"
                r="9.24875"
              />
            </g>
          </svg>
          <span
            className="ml-2"
          >
            Test2
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <footer
    className="mt-auto"
  >
    <div
      className="flex items-center gap-3 p-4 pt-8 text-white"
      data-testid="nav-bottom-user"
    >
      <img
        alt="Testing title"
        className="h-12 w-12 rounded-md bg-white"
        src="test.jpg"
      />
      <div
        className="text-p-3 overflow-hidden"
      >
        <p
          className="font-bold"
        >
          Testing title
        </p>
        <span
          className="block w-full truncate"
        >
          Testing subtitle
        </span>
      </div>
    </div>
  </footer>
</section>
`;

exports[`<Navbar /> should render correctly 1`] = `
<section
  className="bg-sierra-night-100 flex h-full min-h-screen flex-col "
>
  <div
    className="flex p-5"
  >
    <picture
      className=""
    >
      <svg
        className="text-white"
        fill="none"
        viewBox="0 0 152 40"
        width={106}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M115.218 33.268c-1.532 0-2.482-1.063-2.482-2.46V19.87c0-3.737.827-6.016 2.973-8.082 1.808-1.762 4.904-2.795 7.172-2.643 1.38.091 2.391.973 2.33 2.43-.031 1.429-1.104 2.34-2.483 2.28-1.226-.03-2.636.486-3.464 1.306-1.042 1.002-1.532 2.34-1.532 4.557v11.059c-.031 1.428-.981 2.491-2.514 2.491z"
          fill="currentColor"
        />
        <path
          d="M62.16 29.896c-1.378-4.497-4.014-7.2-6.007-8.78-.95-.76-2.237-1.61-3.586-2.218 1.196-1.064 5.303-4.74 6.008-5.469.98-1.033 1.072-2.491.122-3.494-1.042-1.154-2.482-1.033-3.617-.121-.766.638-5.517 5.225-7.846 7.474V2.522c0-1.428-.981-2.521-2.514-2.521-1.563-.03-2.544 1.063-2.544 2.521v28.194c0 1.428.981 2.522 2.544 2.522 1.563 0 2.514-1.094 2.514-2.522v-8.385c.214 0 .368.03.613.03 2.82.304 7.54 3.22 9.41 9.054.52 1.58 1.992 2.066 3.218 1.701 1.38-.395 2.176-1.701 1.686-3.22z"
          fill="currentColor"
        />
        <path
          d="M83.923 9.328c-1.379-.578-2.666-.03-3.218 1.306-.061.122-4.169 9.844-6.682 15.494l-7.05-15.555c-.552-1.215-1.93-1.883-3.34-1.245-1.38.638-1.779 2.005-1.258 3.311l8.92 19.292c-.766 1.367-1.257 1.914-1.931 2.34-.49.303-1.073.516-2.697.516-1.318-.03-2.269.85-2.269 2.37 0 1.489 1.135 2.34 2.453 2.4 2.084.122 3.74-.243 5.21-1.185 2.177-1.367 3.127-3.342 4.66-6.532 1.44-3.008 8.153-18.563 8.46-19.292.612-1.306.122-2.643-1.258-3.22z"
          fill="currentColor"
        />
        <path
          d="M109.027 17.379c-.981-4.74-5.517-8.416-10.881-8.416-6.1 0-11.556 4.74-11.556 12.244 0 3.402 1.195 6.592 3.341 8.84 2.115 2.219 5.303 3.373 8.521 3.373 2.759 0 5.425-.668 7.295-1.914 1.104-.73 1.533-1.975.92-3.16-.705-1.367-2.085-1.64-3.249-.972-1.533.82-2.912 1.489-5.027 1.489-2.973 0-5.487-1.762-6.468-4.83 3.587-.7 12.384-2.431 14.315-2.826 2.145-.486 3.218-1.854 2.789-3.828zm-17.38 2.126c.552-3.463 2.82-5.985 6.53-5.985 2.543 0 4.72 1.459 5.67 3.585l-12.2 2.4z"
          fill="currentColor"
        />
        <path
          d="M139.433 8.75c-7.08 0-12.567 5.439-12.567 12.457 0 7.018 5.487 12.456 12.567 12.456S152 28.225 152 21.207 146.513 8.75 139.433 8.75zm0 20.295c-4.261 0-7.571-3.403-7.571-7.838 0-4.436 3.28-7.839 7.571-7.839 4.261 0 7.571 3.403 7.571 7.839 0 4.405-3.31 7.838-7.571 7.838z"
          fill="currentColor"
        />
        <path
          d="M17.594 9.176a8.992 8.992 0 00-4.506 1.215c.399 1.246.613 2.613.613 3.98 0 .73-.061 1.458-.184 2.157a15.454 15.454 0 015.67 9.995 8.682 8.682 0 007.235-8.567c0-4.861-3.955-8.78-8.828-8.78z"
          fill="#F5B333"
        />
        <path
          d="M11.127 2.036c.183-.334.46-.486 1.042-.699.582-.212.888-.243 1.226-.06.337.152.49.364.674.88.123.396.95 3.403 1.042 3.98.184.79-.214 1.277-.889 1.49-.674.242-1.256.06-1.594-.7-.245-.546-1.379-3.463-1.532-3.858-.123-.456-.123-.729.03-1.033z"
          fill="#F5B333"
        />
        <path
          d="M21.793 1.307c.337-.152.675-.121 1.257.061.582.182.858.334 1.042.668.184.335.184.578 0 1.094-.153.395-1.257 3.312-1.532 3.828-.338.73-.92.912-1.594.699-.675-.213-1.073-.668-.859-1.489.123-.577.92-3.615 1.043-4.01.183-.486.337-.699.643-.85z"
          fill="#F5B333"
        />
        <path
          d="M30.866 6.928c.368.06.613.303.98.79.369.486.491.76.46 1.124-.06.364-.214.577-.643.911-.337.243-2.973 1.944-3.494 2.218-.705.395-1.288.213-1.686-.365-.43-.577-.46-1.154.184-1.701.46-.395 2.88-2.37 3.218-2.643.398-.304.644-.395.98-.334z"
          fill="#F5B333"
        />
        <path
          d="M34.881 16.771c.276.274.307.577.337 1.215.031.608-.06.912-.306 1.185-.245.274-.49.334-1.073.365-.429 0-3.555-.183-4.168-.243-.828-.091-1.165-.578-1.134-1.276 0-.699.306-1.215 1.164-1.276.613-.061 3.74-.243 4.169-.274.49-.03.766.061 1.011.304z"
          fill="#F5B333"
        />
        <path
          d="M32.276 27.04c.03.365-.092.669-.46 1.155-.337.516-.582.698-.95.79-.368.06-.613-.03-1.073-.335-.337-.243-2.79-2.217-3.218-2.612-.613-.547-.583-1.155-.153-1.702.398-.577.98-.79 1.685-.364.521.303 3.188 1.974 3.525 2.217.43.304.583.517.644.851z"
          fill="#F5B333"
        />
        <path
          d="M24.061 33.846c-.184.334-.46.486-1.042.698-.582.213-.889.243-1.226.061-.337-.152-.49-.364-.674-.881-.123-.395-.95-3.403-1.042-3.98-.184-.79.214-1.276.888-1.489.675-.243 1.257-.06 1.594.7.246.546 1.41 3.463 1.533 3.858.153.455.123.729-.03 1.033z"
          fill="#F5B333"
        />
        <path
          d="M11.617 17.288c.245-.942.368-1.914.368-2.887 0-4.192-2.269-7.838-5.67-9.843a17.302 17.302 0 00-1.87 1.792c3.003 1.489 5.057 4.527 5.057 8.051 0 .547-.061 1.094-.153 1.641-.521-.213-1.042-.425-1.594-.577.061-.335.092-.699.092-1.033 0-3.008-1.84-5.59-4.475-6.745a18.57 18.57 0 00-1.318 2.096c1.93.669 3.34 2.492 3.34 4.649 0 .182 0 .364-.03.547a13.635 13.635 0 00-1.594-.091h-.092c.03-.152.03-.304.03-.456a3.204 3.204 0 00-2.39-3.1c-.307.73-.582 1.49-.766 2.28.398.06.735.394.735.82 0 .455-.367.85-.858.85-.092 0-.153 0-.214-.03 0 .03 0 .091-.031.122-.03.121-.03.212-.03.334 0 .06-.031.121-.031.182A14.694 14.694 0 000 17.956v.03a11.047 11.047 0 013.77-.668c6.253 0 11.341 5.043 11.341 11.241 0 2.309-.705 4.466-1.93 6.259.857.212 1.716.364 2.635.455a13.51 13.51 0 001.778-6.744c0-4.649-2.39-8.78-5.977-11.241z"
          fill="#9FACD9"
        />
        <path
          d="M3.77 19.445a9.481 9.481 0 00-3.617.729c.123.85.276 1.67.49 2.46a6.692 6.692 0 013.127-.759c3.74 0 6.774 3.008 6.774 6.715a6.477 6.477 0 01-1.747 4.466c.736.425 1.471.79 2.268 1.093 1.196-1.55 1.931-3.463 1.931-5.56-.03-5.043-4.138-9.144-9.226-9.144z"
          fill="#9FACD9"
        />
        <path
          d="M3.77 24.002a4.48 4.48 0 00-2.39.668c.306.76.674 1.489 1.103 2.157a2.178 2.178 0 011.287-.425c1.196 0 2.176.972 2.176 2.157a2.08 2.08 0 01-.827 1.671 23.14 23.14 0 001.839 1.61 4.57 4.57 0 001.41-3.28c0-2.522-2.054-4.558-4.598-4.558z"
          fill="#9FACD9"
        />
      </svg>
    </picture>
  </div>
  <nav>
    <ul
      className="text-white"
    >
      <li
        className="flex items-center"
        data-testid="nav-element"
      >
        <div
          className="w-full px-4 py-3 flex hover:bg-sierra-night-80 focus:bg-sierra-night-80 bg-sierra-night-80"
          data-testid="dummy-link"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.5714 11.2857H14.8571C14.4626 11.2857 14.1428 11.6055 14.1428 12V20.5715C14.1428 20.9659 14.4626 21.2857 14.8571 21.2857H20.5714C20.9659 21.2857 21.2857 20.9659 21.2857 20.5715V12C21.2857 11.6055 20.9659 11.2857 20.5714 11.2857Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
            <path
              d="M20.5714 2.71429H14.8571C14.4626 2.71429 14.1428 3.03409 14.1428 3.42858V6.30001C14.1428 6.6945 14.4626 7.01429 14.8571 7.01429H20.5714C20.9659 7.01429 21.2857 6.6945 21.2857 6.30001V3.42858C21.2857 3.03409 20.9659 2.71429 20.5714 2.71429Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
            <path
              d="M9.14293 2.71429H3.42864C3.03415 2.71429 2.71436 3.03409 2.71436 3.42858V12C2.71436 12.3945 3.03415 12.7143 3.42864 12.7143H9.14293C9.53742 12.7143 9.85721 12.3945 9.85721 12V3.42858C9.85721 3.03409 9.53742 2.71429 9.14293 2.71429Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
            <path
              d="M9.14293 16.9857H3.42864C3.03415 16.9857 2.71436 17.3055 2.71436 17.7V20.5714C2.71436 20.9659 3.03415 21.2857 3.42864 21.2857H9.14293C9.53742 21.2857 9.85721 20.9659 9.85721 20.5714V17.7C9.85721 17.3055 9.53742 16.9857 9.14293 16.9857Z"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
          </svg>
          <span
            className="ml-2"
          >
            Test1
          </span>
        </div>
      </li>
      <li
        className="flex items-center"
        data-testid="nav-element"
      >
        <a
          className="w-full px-4 py-3 flex hover:bg-sierra-night-80 focus:bg-sierra-night-80"
          href="/"
        >
          <svg
            fill="none"
            height={19}
            viewBox="0 0 19 20"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="9.24875"
              cy="9.24875"
              fill="#232559"
              r="9.24875"
            />
            <mask
              height="17"
              id="mask0_22_2320"
              maskUnits="userSpaceOnUse"
              style={
                {
                  "maskType": "alpha",
                }
              }
              width="14"
              x="2"
              y="3"
            >
              <path
                d="M12.4645 6.90497C12.4645 8.73556 11.0101 10.9933 9.13163 10.9933C7.25311 10.9933 5.85938 8.73556 5.85938 6.90497C5.85938 5.07438 7.31371 3.54889 9.19222 3.54889C11.0707 3.54889 12.4645 5.01336 12.4645 6.90497Z"
                fill="#B3B3B3"
              />
              <path
                d="M15.4338 16.546C15.4338 16.7901 12.1615 19.0479 9.13165 19.0479C6.22298 18.9868 2.89014 16.9122 2.89014 16.546C2.89014 13.8002 6.46537 11.6035 9.19225 11.6035C11.9191 11.6035 15.4338 13.8002 15.4338 16.546Z"
                fill="#B3B3B3"
              />
            </mask>
            <g
              mask="url(#mask0_22_2320)"
            >
              <circle
                cx="9.24863"
                cy="9.24881"
                fill="#EDEDF0"
                r="9.24875"
              />
            </g>
          </svg>
          <span
            className="ml-2"
          >
            Test2
          </span>
        </a>
      </li>
    </ul>
  </nav>
</section>
`;
